components/BottomSheet/src/MDCBottomSheetTransitionController.m

```diff
@@ -51,69 +50,28 @@ @implementation MDCBottomSheetTransitionController
 
 - (NSTimeInterval)transitionDuration:
     (nullable __unused id <UIViewControllerContextTransitioning>)transitionContext {
-  return MDCBottomSheetTransitionDuration;
+  return MDCBottomSheetMotionSpec.transitionDuration;
 }
 
 - (void)animateTransition:(id <UIViewControllerContextTransitioning>)transitionContext {
-  // If a view in the transitionContext is nil, it likely hasn't been loaded by its ViewController
-  // yet.  Ask for it directly to initiate a loadView on the ViewController.
-  UIViewController *fromViewController =
-      [transitionContext viewControllerForKey:UITransitionContextFromViewControllerKey];
-  UIView *fromView = [transitionContext viewForKey:UITransitionContextFromViewKey];
-  if (fromView == nil) {
-    fromView = fromViewController.view;
-  }
-
-  UIViewController *toViewController =
-  [transitionContext viewControllerForKey:UITransitionContextToViewControllerKey];
-      UIView *toView = [transitionContext viewForKey:UITransitionContextToViewKey];
-  if (toView == nil) {
-    toView = toViewController.view;
-  }
-
-  UIViewController *toPresentingViewController = toViewController.presentingViewController;
-  BOOL presenting = (toPresentingViewController == fromViewController) ? YES : NO;
+  // We're only implementing UIViewControllerAnimatedTransitioning so that we can customize the
+  // transition's duration - all of the actual animation logic happens in
+  // MDCBottomSheetPresentationController (which can't customize the duration). In order to make
+  // UIKit believe that we're actually animating something, we create a temporary bogus view and
+  // animate it for the desired duration. This ensures that UIKit's default animations animate
+  // alongside our bogus animation (and therefor the corresponding duration).
```

Comment:
If we didn't want to control the transition's duration we could get by without providing an animation controller altogether, but unfortunately we do want to tighten up the transition's duration. In general most transitions do have some amount of animation that occurs during the transition portion so we'd implement an animation controller accordingly, but in this case the entire transition is managed by the presentation controller (which is not as common).

Reply:
I would expect the presentation controller to grab the duration from the transitionCoordinator in its presentationTransitionWillBegin.
I would expect animateTransition: to handle adding the presented view controller's view to the hierarchy / it's containerView..

Reply:
> I would expect the presentation controller to grab the duration from the transitionCoordinator in its presentationTransitionWillBegin.

The presentation controller doesn't need to grab the transition duration because it uses the `animateAlongsideTransition` APIs.

Reply:
> I would expect animateTransition: to handle adding the presented view controller's view to the hierarchy / it's containerView..

UIKit appears to be doing the view hierarchy management for us which is why I removed the code.

Reply:
> I would expect animateTransition: to handle adding the presented view controller's view to the hierarchy / it's containerView..

Done.